src/bosh_azure_cpi/lib/cloud/azure/vms/vm_manager.rb

```diff
@@ -83,13 +89,13 @@ def create(instance_id, location, stemcell_id, vm_props, network_configurator, e
       diagnostics_storage_account = task_get_diagnostics_storage_account.value!
 
       # create vm params
-      vm_params = {
+      vm_params.merge!(
```

Comment:
weird to merge instance_id together with original vm_params but don't use it, see line 154.
And vm_params is for azure_client, and I suggest not to merge instance_id to it. If you want to use instance_id outside, just return two values: `return instance_id, vm_params`.

Reply:
yes, indeed.